### DISCUSSION TITLE
support a new API for stat info without file content？

### DISCUSSION THREAD

**Discussion by rowthan** [coddingtonbear/obsidian-local-rest-api#92] [Ideas]

**Comment by coddingtonbear** [comment]
I believe this exists via setting the `Accept` header to `application/vnd.olrapi.note+json`?  You can see more in the [online docs](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Active%20File/get_active_) around each of the `GET` APIs,

**Reply by rowthan** [reply]
it works but response the content still. i dont think it is good in same case.
such as checking the file whether exist or get the meta info of the file.
maybe its a waste.

**Reply by coddingtonbear** [reply]
I believe you’re mistaken!  If you send that `Accept` header, you will receive only the metadata, no?

**Reply by rowthan** [reply]
I think I have not made myself understood.
what i want is that don't response the `content`!  because i just want get the meta info, i don't care what the content is.
The usage scenario is that i want to quickly obtain the original information of all files to decide whether they need to be updated depend on `ctime` or `utime`

now you have response all infos both metadata and content. i think it's a waste.

so I suggest create a new api just for retrieving meta info  for single file even  list of files.

**Reply by rowthan** [reply]
![inner](https://github.com/coddingtonbear/obsidian-local-rest-api/assets/12638456/9048ec5d-b00f-4687-8bfb-02d5d3b4766b)

don't response this field.

**Reply by coddingtonbear** [reply]
Oh, you're right -- we do send the file content in the metadata!  I'd be very interested in seeing some numbers around how much such a change would impact performance -- files in folks' vaults are in general very small in the context of file sizes on the internet.  For example, the median and average file sizes in _my_ vault are just 566 bytes and 1076 bytes respectively.  Additionally, this is a *local* API, so these files don't even need to cross a network.  I'd guess that the transmission cost would be dwarfed by other overhead involved in processing the request.

If it did turn out that we were able to prove that such a change wouldn't be a premature optimization: I'm not sure I'm willing to support an entirely new API for sending a metadata response without that content, but I'd consider a PR that added a new `Accept` header option that excluded that `content` field.  If you're looking for insight into how you might do that; you'd need to do a couple things:

1. Add a new `Accept` header option here:
2. Add a new conditional block here that checked for the new header and returned only the metadata without content: https://github.com/coddingtonbear/obsidian-local-rest-api/blob/adfab3d51970aa7e594e6d721c43719e0bdf527a/src/requestHandler.ts#L236 -- you'll probably want to update `getFileMetadataObject` such that it can conditionally return the metadata object with content so that.
3. Update docs to describe that `Accept` header's purpose

Hope this helps!
